src/write.ts

```diff
@@ -148,11 +148,56 @@ function floatStr(n: number) {
     return n.toString();
 }
 
+const round = (n: number, places: number = 3): string =>
+    n.toFixed(places).replace(/\.?0+$/, "");
+
+function strDuration(n: number): string {
+    let unit = "ns";
+    // I could have DRYd this with some defunctionalization
+    // and a for loop, but I really don't care enough.
+    if (n > 1000) {
+        unit = "μs";
+        n /= 1000;
+    }
+    if (n > 1000) {
+        unit = "ms";
+        n /= 1000;
+    }
+    if (n > 1000) {
+        unit = "s";
+        n /= 1000;
+    }
+    if (n > 60) {
+        const mins = Math.floor(n / 60);
+        const secs = n % 60;
+        return `${mins}m ${round(secs)}s`
+    }
+    return `${round(n)}${unit}`;
```

Comment:
This might be because I'm not very fluent in Javascript, but it says `round(n)` here whereas [the demo you linked](https://github.com/414owen/outlines-core/actions/runs/13232874632) shows a duration like "1.726μs"?

Reply:
Round is a badly named function I committed in this pr 🙈

I'll rename it this afternoon

Reply:
I noticed it now, sorry, my bad :facepalm: 
I confused it with `Math.round`. The function itself is fine :+1: